src/rp2040_flash.rs

```diff
@@ -42,3 +45,43 @@ pub fn read_device_config_from_rp2040_flash() -> &'static [u8] {
     let addr = (FLASH_XIP_BASE + FLASH_END - FLASH_USER_SIZE) as *const u8;
     unsafe { slice::from_raw_parts(addr, FLASH_USER_SIZE as usize) }
 }
+
+pub fn read_is_audio_from_rp2040_flash() -> bool {
```

Comment:
I'd rather see the calling code reading out this calling something like `DeviceConfig::load_existing_config_from_flash().map(|config| config.config().is_audio_device)` - so that we're not hard-coding the assumption about there the audio flag is in the bytes here.

Reply:
this causes memory problems otherwise